pydata_sphinx_theme/_templates/version-switcher.html

```diff
@@ -0,0 +1,79 @@
+<div class="dropdown" id="version_switcher">
+    <button type="button" class="btn btn-primary btn-sm navbar-btn dropdown-toggle" id="version_switcher_button" data-toggle="dropdown">
+        {{ theme_switcher.get('version_match') }}  <!-- this text may get changed later by javascript -->
+        <span class="caret"></span>
+    </button>
+    <div id="version_switcher_menu" class="dropdown-menu list-group-flush py-0" aria-labelledby="version_switcher_button">
+    <!-- dropdown will be populated by javascript on page load -->
+    </div>
+</div>
+
+<!-- NOTE: this JS must live here (not in our global JS file) because it relies
+     on being processed by Jinja before it is run (specifically for replacing
+     variables {{ pagename }} and {{ theme_switcher }}.
+-->
```

Comment:
I think we should somehow address this one before merging the PR, @drammock.

We should try to move most of the JS to the `src/js/index.js` file. In general, I believe we should minimize the amount of JS code in the template as far as we can. And we can borrow some ideas from #433 to execute that idea. You can see they are only adding JS code in the template to "persist" the information about `pagename` and `theme_swticher`. They are doing that using the [window object](https://github.com/pydata/pydata-sphinx-theme/pull/433/files#diff-7368bf1d94bd6db55fb4980a4d366123b09d0a8cf674d983e3a920e2b7412b1eR2-R9). We may use the same approach or maybe some other ones such as [`data-*` attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) that could be more specific? ones (instead of polluting the window namespace) and it might play better with the template...
But the important piece, whatever implementation you think about, is the idea/aim to minimize the JS code in the template and group that code in the "natural" JS place we currently have, IMHO.

@drammock, I know this is a long-lasting PR but would you be willing to make this change?

Reply:
like many (most?) contributors here, I am not a JS developer, I am a python developer who knows just enough JS to get the job done, and figure out a lot as I go along.  I invite/encourage you to push commits to this branch if you have a clear idea of how to do what you suggest.  Otherwise I can say that I will *intend* to do as you suggest, but offer no promises as to the timeline because I have no way of estimating how long it will take me to figure it out / get it right, which means I'm less likely to try to squeeze it in when I have just a few minutes to spare.

If that's not acceptable, another option is to merge now (once merge conflict is fixed, I see this PR has again gone stale) and migrate the JS in a follow-up PR, either by me (when I get around to it) or by anyone else who is inspired.

Reply:
Generally I think it would indeed be good to have our JS in `src/js/index.js`, but if that makes this quite a bit more complicated, I also think that practicality beats purity in this case. 

Now, from the explanation above, it seems there is a possible solution already. But at the same time, this is in the end also only an implementation detail, not impacting in any way how the end user uses/configures this. So I would say that this therefore is also well suited as a follow-up (given how long this already has taken).

Reply:
Thank you for the feedback @drammock and @jorisvandenbossche!
A follow-up PR seems sensible to me (btw, I had to ask you first @drammock, in case you wanted to do it 😉 ).